File: ParaView/GUI/Widgets/Examples/Cxx/HelloWorld/KWHelloWorldExample.cxx
#include "vtkKWApplication.h"
#include "vtkKWWindowBase.h"
#include "vtkKWLabel.h"
#include "vtkKWFrame.h"

#include <vtksys/SystemTools.hxx>
#include <vtksys/CommandLineArguments.hxx>

int my_main(int argc, char *argv[])
{
  // Initialize Tcl

  Tcl_Interp *res = vtkKWApplication::InitializeTcl(argc, argv, &cerr);
  if (!res)
    {
    cerr << "Error: InitializeTcl failed" << endl ;
    return 1;
    }

  // Process some command-line arguments
  // The --test option here is used to run this example as a non-interactive 
  // test for software quality purposes. You can ignore it.

  int option_test = 0;
  vtksys::CommandLineArguments args;
  args.Initialize(argc, argv);
  args.AddArgument(
    "--test", vtksys::CommandLineArguments::NO_ARGUMENT, &option_test, "");
  args.Parse();
  
  // Create the application
  // If --test was provided, ignore all registry settings, and exit silently
  // Restore the settings that have been saved to the registry, like
  // the geometry of the user interface so far.

  vtkKWApplication *app = vtkKWApplication::New();
  app->SetName("KWHelloWorldExample");
  if (option_test)
    {
    app->SetRegistryLevel(0);
    app->PromptBeforeExitOff();
    }
  app->RestoreApplicationSettingsFromRegistry();

  // Set a help link. Can be a remote link (URL), or a local file

  // vtksys::SystemTools::GetFilenamePath(__FILE__) + "/help.html";
  app->SetHelpDialogStartingPage("http://www.kwwidgets.org");

  // Add a window
  // Set 'SupportHelp' to automatically add a menu entry for the help link

  vtkKWWindowBase *win = vtkKWWindowBase::New();
  win->SupportHelpOn();
  app->AddWindow(win);
  win->Create();

  // Add a label, attach it to the view frame, and pack
  
  vtkKWLabel *hello_label = vtkKWLabel::New();
  hello_label->SetParent(win->GetViewFrame());
  hello_label->Create();
  hello_label->SetText("Hello, World!");
  app->Script("pack %s -side left -anchor c -expand y", 
              hello_label->GetWidgetName());
  hello_label->Delete();

  // Start the application
  // If --test was provided, do not enter the event loop and run this example
  // as a non-interactive test for software quality purposes.

  int ret = 0;
  win->Display();
  if (!option_test)
    {
    app->Start(argc, argv);
    ret = app->GetExitStatus();
    }
  win->Close();

  // Deallocate and exit

  win->Delete();
  app->Delete();
  
  return ret;
}

#ifdef _WIN32
#include <windows.h>
int __stdcall WinMain(HINSTANCE, HINSTANCE, LPSTR lpCmdLine, int)
{
  int argc;
  char **argv;
  vtksys::SystemTools::ConvertWindowsCommandLineToUnixArguments(
    lpCmdLine, &argc, &argv);
  int ret = my_main(argc, argv);
  for (int i = 0; i < argc; i++) { delete [] argv[i]; }
  delete [] argv;
  return ret;
}
#else
int main(int argc, char *argv[])
{
  return my_main(argc, argv);
}
#endif

File: GUI/Widgets/Examples/Python/HelloWorld/KWHelloWorldExample.py
# Load the KWWidgets package

from kwwidgets import *

# Load requires system modules

import sys

# Process some command-line arguments
# The --test option here is used to run this example as a non-interactive test
# for software quality purposes. Ignore this feature in your own application.

if "--test" in sys.argv:
    option_test = 1
else:
    option_test = 0

def main(argv):
    
    # Initialize Tcl

    vtkKWApplication.InitializeTcl(len(argv), argv)

    # Create the application
    # If --test was provided, ignore all registry settings, and exit silently
    # Restore the settings that have been saved to the registry, like
    # the geometry of the user interface so far.

    app = vtkKWApplication()
    app.SetName("KWHelloWorldExample")
    if option_test:
        app.SetRegistryLevel(0)
        app.PromptBeforeExitOff()

    app.RestoreApplicationSettingsFromRegistry()
    
    # Set a help link. Can be a remote link (URL), or a local file

    app.SetHelpDialogStartingPage("http://www.kwwidgets.org")

    # Add a window
    # Set 'SupportHelp' to automatically add a menu entry for the help link

    win = vtkKWWindowBase()
    win.SupportHelpOn()
    app.AddWindow(win)
    win.Create()

    # Add a label, attach it to the view frame, and pack

    hello_label = vtkKWLabel()
    hello_label.SetParent(win.GetViewFrame())
    hello_label.Create()
    hello_label.SetText("Hello, World!")
    app.Script("pack %s -side left -anchor c -expand y",
               hello_label.GetWidgetName())

    # Start the application
    # If --test was provided, do not enter the event loop and run this example
    # as a non-interactive test for software quality purposes.

    ret = 0
    win.Display()
    if not option_test:
        app.Start(len(sys.argv), sys.argv)
        ret = app.GetExitStatus()

    win.Close()

    sys.exit(ret)

if __name__ == "__main__":
    main(sys.argv)
